src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs

```diff
@@ -112,36 +112,23 @@ private void OnBeginExecute(EventWrittenEventArgs eventData)
             return;
         }
 
+        string dataSource = (string)eventData.Payload[1];
+        string databaseName = (string)eventData.Payload[2];
+        var startTags = SqlActivitySourceHelper.GetTagListFromConnectionInfo(dataSource, databaseName, this.options, out var activityName);
         var activity = SqlActivitySourceHelper.ActivitySource.StartActivity(
-            SqlActivitySourceHelper.ActivityName,
+            activityName,
             ActivityKind.Client,
             default(ActivityContext),
-            SqlActivitySourceHelper.CreationTags);
+            startTags);
```

Comment:
Interesting that `StartActivity` doesn't have an overload which accepts `TagList`. We should ask @tarekgh for one 🤣

In the meantime, maybe don't use `TagList` at all. Because this will box it up.

A `ThreadStatic` is probably fine we did that here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/0351a0bcbcf8c43bcaab1fb17a01a9b845ca67c2/src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/ActivityHelper.cs#L30-L31

Reply:
Yea, I was bummed there was no `StartActivity` with `TagList` as well 😢.

> A ThreadStatic is probably fine we did that here

I had a similar thought to use ThreadStatic. Ultimately, these tags will need to be shared by the metric instrumentation as well. A ThreadStatic would allow us to capture them on command start and then use them on command end when recording the metric. I'll try this in a follow up.

In the meantime, I'll change this to not use TagList for this PR.

Reply:
> Interesting that StartActivity doesn't have an overload which accepts TagList. We should ask @tarekgh for one 🤣

You may update https://github.com/dotnet/runtime/issues/103245#issuecomment-2189834442 with whatever request. the runtime issue was talking about CreateActivity but can incluse StartActivity too.